todo-app/src/Stores/TasksListStore.js

```diff
@@ -0,0 +1,151 @@
+import {action,configure,observable} from 'mobx'
+import moment from 'moment'
+
+import {loadFromLocalStorage,saveToLocalStorage} from '../localStorage'
+import todoList from '../todoList.json'
+
+configure({
+    enforceActions: true
+})
+
+
+
+class TasksListStore {
+
+    @observable
+    todoList = loadFromLocalStorage() || todoList
+
+
+    @observable
+    selectedToEdit = null
+
+    @observable
+    editedTask = ''
+
+    @observable
+    newTask = {
+        description: '',
+        date: '',
+    }
+
+    @action
+    selected = (id) => {
+        const todoItems = this.todoList.map(task => {
+            if (task.id === id) {
+                return {
+                    "id": task.id,
+                    "description": task.description,
+                    "deadline": task.deadline,
+                    "selected": !task.selected,
+                    "isDone": task.isDone,
+                    "timeout": task.timeout
+                }
+            } else {
+                return task
+            }
+        })
+        this.todoList = todoItems
+        saveToLocalStorage(this.todoList)
+    }
+
+    @action
+    setDone = (id) => {
+        const todoItems = this.todoList.map(task => {
+            if (task.id === id) {
+                return {
+                    "id": task.id,
+                    "description": task.description,
+                    "deadline": task.deadline,
+                    "selected": task.selected,
+                    "isDone": !task.isDone,
+                    "timeout": task.timeout
+                }
+            } else {
+                return task
+            }
+        })
+        this.todoList = todoItems
+        saveToLocalStorage(this.todoList)
+    }
+
+    @action
+    addNewTask = (newTask, newDeadLine) => {
+
+        const todoItems = this.todoList
+        const id = this.todoList.map(task => task.id)
+        const newId = this.todoList.length > 0 ? Math.max(...id) + 1 : 1
+        todoItems.push({
+            "id": newId,
+            "description": newTask.toUpperCase(),
+            "deadline": newDeadLine,
+            "selected": false,
+            "isDone": false,
+            "timeout": this.setTimeout(newDeadLine)

```

Comment:
You should definitely choose a new name for `this.setTimeout`. `setTimeout` is already a built-in method to JavaScript, and does something very very different from your class method here.